src/pages/api/repair-request/index.ts

```diff
@@ -32,5 +34,11 @@ async function createRepairRequest(
     }
   });
 
-  return res.status(200).json({ id: record.id });
+  const requestId = record.id;
+
+  // send confirmation email to customer
+  const emailSubject = "Repaire Request Confimation";
+  await sendEmail(emailSubject, userId!, requestId); // TODO: Do not handle sending result for now
```

Comment:
not sure if you need to await this function @justinliangg could you advice on this? based on your function below

if needed a try catch should be added

Reply:
Good point. I think I can remove await because we do not handle the response for now.

Reply:
After moving the `await` here, I tested the service again in the following scenarios:
1. Created a repair request via the web page in mock mode and received a mock email.
2. Created a repair request via the web page in product mode and received a real email.
3. Created a repair request via Vitest in mock mode but did not receive a mock email.
4. Created a repair request via Vitest in product mode but did not receive a real email.

I guess there may be something associated with Vitest. After the tested endpoint returns, Vitest does not continue the background job and just returns a pass. To address this, I added a setTimeout() in the test code, which puts the vitest thread to sleep for 8 seconds and then executes the except().

After adding the setTimeout(), I could:
1. Created a repair request via Vitest in mock mode and received a mock email.
2. Created a repair request via Vitest in product mode and received a real email.

However, I'm not sure if this behavior is specific to Vitest or if I need to add some configuration to change Vitest to make it send an email after returning a pass. 

Do you guys have any better solutions for this Vitest issue? @dct0 @BoboJeager @justinliangg

Reply:
Yeah, I am not really sure what is going on with the Vitest. I have encountered this same issue before using Jest and couldn't really find a solution to it other than making sure I `await` the send email function. 

If you still want to try, maybe see if you can flush all the promises first. https://stackoverflow.com/questions/44741102/how-to-make-jest-wait-for-all-asynchronous-code-to-finish-execution-before-expec